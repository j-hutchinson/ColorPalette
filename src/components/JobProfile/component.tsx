import React from 'react';
import {
    StyledCompanyName,
    StyledTitle,
    StyledTop,
    StyledDate
} from './styles';

const JobProfile = ({ name, startDate, endDate = "present", title }) => (
    <div>
        <StyledTop>
            <StyledCompanyName>{name}</StyledCompanyName>
            <StyledDate>{startDate} {"->"} {endDate}</StyledDate>
        </StyledTop>
        <StyledTitle>{title}</StyledTitle>
    </div>
)

export default JobProfile;