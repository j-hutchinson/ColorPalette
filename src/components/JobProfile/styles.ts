import styled from 'styled-components'

export const StyledContainer = styled.div`
    margin: 32px 0;
`;

export const StyledCompanyName = styled.h3`
    margin: 0px;
    font-size: 22px;
    display: inline;
`;
export const StyledTitle = styled.h4`
    margin: 8px 0px 16px;
    color: #878787;
    font-size: 18px;
    font-style: normal;
`;
export const StyledTop = styled.div`
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    font-style: normal;
`;

export const StyledDate = styled.p`
    display: inline;
    font-size: 18px;
    margin: 4px 0px 0px;
`;

export const StyledDescription = styled.p`
    display: block;
    font-size: 18px;
    width: 80%;
    line-height: 22px;
    margin: 4px 0px 0px;
`;
