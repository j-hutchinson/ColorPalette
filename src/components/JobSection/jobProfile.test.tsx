import { shallow } from 'enzyme';
import JobProfile from './component';

describe('JobProfile component', () => {
    const profile = {
        desc: "desc",
        endDate: "endDate",
        startDate: "startDate",
        name: "name",
        title: "title",
        link: 'somewhere_over_the_rainbow'
    }
    test('component matches snapshot with full profile', () => {
        expect.assertions(1);

        const wrapper = shallow(<JobProfile {...profile} />);

        expect(wrapper).toMatchSnapshot();
    });
    test('component matches snapshot with no endDate', () => {
        expect.assertions(1);

        const wrapper = shallow(<JobProfile {...profile} endDate={null} />);

        expect(wrapper).toMatchSnapshot();
    });
});