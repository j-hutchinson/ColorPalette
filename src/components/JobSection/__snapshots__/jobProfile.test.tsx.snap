// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobProfile component component matches snapshot with full profile 1`] = `
<styled.section>
  <div>
    <styled.h3>
      name
    </styled.h3>
    <styled.p>
      startDate
       
      -&gt;
       
      endDate
    </styled.p>
    <styled.h4>
      title
    </styled.h4>
    <styled.p>
      desc
    </styled.p>
  </div>
  <styled.div />
</styled.section>
`;

exports[`JobProfile component component matches snapshot with no endDate 1`] = `
<styled.section>
  <div>
    <styled.h3>
      name
    </styled.h3>
    <styled.p>
      startDate
       
      -&gt;
       
    </styled.p>
    <styled.h4>
      title
    </styled.h4>
    <styled.p>
      desc
    </styled.p>
  </div>
  <styled.div />
</styled.section>
`;
